src/app/components/movie-details/movie-details.component.ts

```diff
@@ -4,52 +4,112 @@ import {
   NgIf
 } from '@angular/common';
 import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
   Component,
   Input,
-  OnDestroy
+  OnDestroy,
+  OnInit
 } from '@angular/core';
-import { Subscription } from 'rxjs';
+import {
+  Subscription,
+  tap
+} from 'rxjs';
 import { ApiRequestType } from '../../shared/enums/api-request';
+import { LocalStorageKeys } from '../../shared/enums/local-storage';
 import { MediaType } from '../../shared/enums/media-types';
-import { MovieDetails } from '../../shared/interfaces/movies';
+import { RatedCard } from '../../shared/interfaces/general';
+import {
+  MovieDetails,
+  MovieRating,
+} from '../../shared/interfaces/movies';
 import { ImagePathPipe } from '../../shared/pipes/image-path.pipe';
+import { RatingPipe } from '../../shared/pipes/rating.pipe';
 import { ApiService } from '../../shared/services/api.service';
-import { GuestSessionService } from '../../shared/services/guest-session.service';
+import { LocalStorageService } from '../../shared/services/local-storage.service';
+import { UserRateService } from '../../shared/services/user-rate.service';
 import { MovieRatingComponent } from '../movie-rating/movie-rating.component';
 
 @Component({
   selector: 'tmbd-movie-details',
   standalone: true,
-  imports: [ImagePathPipe, NgIf, NgForOf, DatePipe, MovieRatingComponent],
+  imports: [
+    ImagePathPipe,
+    NgIf,
+    NgForOf,
+    DatePipe,
+    MovieRatingComponent,
+    RatingPipe
+  ],
   templateUrl: './movie-details.component.html',
-  styleUrls: ['./movie-details.component.scss']
+  styleUrls: ['./movie-details.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class MovieDetailsComponent implements OnDestroy {
+export class MovieDetailsComponent implements OnInit, OnDestroy {
   @Input() public details?: MovieDetails;
   @Input() public pageType: MediaType.Movie | MediaType.Tv = MediaType.Movie;
 
-  private guestSession = this.guestSessionService.guestSession;
+  private guestSessionValue = this.userRate.guestSession;
+  private ratedList: RatedCard[] = [];
   private rateSubscription!: Subscription;
-  public isVoted = false;
+  public userRating: number | null = null;
 
   constructor(
     private readonly apiService: ApiService,
-    private readonly guestSessionService: GuestSessionService,
+    private readonly userRate: UserRateService,
+    private readonly localStorage: LocalStorageService,
+    private readonly cdr: ChangeDetectorRef,
   ) {}
 
+  public ngOnInit(): void {
+    if (!this.guestSessionValue) {
```

Comment:
All logic with guest session should not be in component. This is not component responsibility to check this value.